scripts/release.sh

```diff
@@ -314,10 +329,70 @@ main() {
     exit 1
   fi
 
-  # TODO: signing process
-  log_warning ""
-  log_warning "WARNING: The release has not been signed and published to github. This must be done manually."
-  log_warning ""
+  if [ "${DRY_RUN}" == "true" ] || [ "${NO_GH_RELEASE}" == 1 ]; then
+    log_warning ""
+    log_warning "WARNING: Skipping creating GitHub release, --no-gh-release is set."
+    log_warning "WARNING: If not running on DRY_MODE, please do the GitHub release manually."
+    log_warning ""
+  else
+    local gh_repo
+    local release_notes_temp_file
+    local release_url
+    local gh_release_args=()
+
+    case "${BRANCH}" in
+      "main")
+        gh_release_args=(--prerelease)
+        ;;
+      "release-3.5")
+        gh_release_args=(--latest)
+        ;;
+    esac
+
+    if [ "${REPOSITORY}" = "$(pwd)" ]; then
+      gh_repo=$(git remote get-url origin)
+    else
+      gh_repo="${REPOSITORY}"
+    fi
+
+    gh_repo=$(echo "${gh_repo}" | sed 's/^[^@]\+@//' | sed 's/https\?:\/\///' | sed 's/\.git$//' | tr ':' '/')
+    log_callout "Creating GitHub release for ${RELEASE_VERSION} on ${gh_repo}"
+
+    release_notes_temp_file=$(mktemp)
+
+    local release_version=${RELEASE_VERSION#v} # Remove the v prefix from the release version (i.e., v3.6.1 -> 3.6.1)
+    local release_version_major_minor=${release_version%.*} # Remove the patch from the version (i.e., 3.6)
+    local release_version_major=${release_version_major_minor%.*} # Extract the major (i.e., 3)
+    local release_version_minor=${release_version_major_minor/*./} # Extract the minor (i.e., 6)
+
+    # Disable sellcheck SC2016, the single quoted syntax for sed is intentional.
+    # shellcheck disable=SC2016
+    sed 's/${RELEASE_VERSION}/'"${RELEASE_VERSION}"'/g' ./scripts/release_notes.tpl.txt |
+      sed 's/${RELEASE_VERSION_MAJOR_MINOR}/'"${release_version_major_minor}"'/g' |
+      sed 's/${RELEASE_VERSION_MAJOR}/'"${release_version_major}"'/g' |
+      sed 's/${RELEASE_VERSION_MINOR}/'"${release_version_minor}"'/g' > "${release_notes_temp_file}"
+
+    if ! gh --repo "${gh_repo}" release view "${RELEASE_VERSION}" &>/dev/null; then
+      maybe_run gh release create "${RELEASE_VERSION}" \
+          --repo "${gh_repo}" \
+          --draft \
+          --title "${RELEASE_VERSION}" \
+          --notes-file "${release_notes_temp_file}" \
+          "${gh_release_args[@]}"
+    fi
+
+    # Upload files one by one, as gh doesn't support passing globs as input.
+    maybe_run find ./release '(' -name '*.tar.gz' -o -name '*.zip' ')' -exec \
+      gh --repo "${gh_repo}" release upload "${RELEASE_VERSION}" {} \;
```

Comment:
Just to double check, is this command idempotent? If we run this command multiple times, will it just replace the files?

Reply:
Thanks for the call out. I just tested and need to add an argument to allow overwriting the file. I'll update the PR shortly.

Reply:
Done in 88d2d2472c32e4cfa0e7223fa0b173865f0ddf3c